### PR TITLE
fix(cost-analysis): refactor and apply costAnalysis route & apply cancellableFetcher

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [x] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
SSIA..
- Add cost-analysis default route for GNB
- Apply `getCancellableFetcher`
- Handle Budget page dataSource issue

### CODE REVIEW

File: apps/web/src/services/cost-explorer/store/cost-query-set-store.ts [hunk 1]
```diff
@@ -29,15 +30,18 @@ export const useCostQuerySetStore = defineStore('cost-query-set', {
     },
     actions: {
         async listCostQuerySets(): Promise<void> {
+            // TODO: apply v2
+            const fetcher = getCancellableFetcher(SpaceConnector.client.costAnalysis.costQuerySet.list);
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/store/cost-query-set-store.ts:34]: suggestion: It doesn't seem necessary to have `fetcher` inside the function.

File: apps/web/src/services/cost-explorer/store/cost-query-set-store.ts [hunk 2]
```diff
@@ -29,15 +30,18 @@ export const useCostQuerySetStore = defineStore('cost-query-set', {
...
+                const { status, response } = await fetcher({
                     data_source_id: this.selectedDataSourceId,
                     query: {
                         filter: [{ k: 'user_id', v: store.state.user.userId, o: 'eq' }],
                     },
                 });
-                this.costQuerySetList = [...managedCostQuerySets, ...results];
+                if (status === 'succeed') {
+                    this.costQuerySetList = [...managedCostQuerySets, ...response.results];
+                }
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/store/cost-query-set-store.ts:44]: nitpick:

```suggestion
                if (status === 'succeed') {
                    this.costQuerySetList = [...managedCostQuerySets, ...response.results];
                } else {
                     this.costQuerySetList = [...managedCostQuerySets];
                }
```

File: apps/web/src/services/cost-explorer/CostExplorerLNB.vue [hunk 3]
```diff
@@ -112,16 +115,18 @@ const state = reactive({
 });
 const dataSourceState = reactive({
     plugins: computed<PluginReferenceMap>(() => allReferenceStore.getters.plugin),
+    dataSourceMap: computed<CostDataSourceReferenceMap>(() => allReferenceStore.getters.allReferenceTypeInfo.costDataSource.referenceMap),
     items: computed<MenuItem[]>(() => {
-        const dataSourceMap: CostDataSourceReferenceMap = allReferenceStore.getters.costDataSource;
+        const dataSourceMap: CostDataSourceReferenceMap = dataSourceState.dataSourceMap;
+        console.debug('dataSourceMap', dataSourceMap);
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/CostExplorerLNB.vue:121]: quibble: 🧐

**Review by yuda110 — APPROVED** [review]
LGTM! I left some comments

**Comment by piggggggggy** [comment]
@yuda110 Thanks~!
I apply your reviews at latest commit!!

File: apps/web/src/services/cost-explorer/CostExplorerLNB.vue [hunk 4]
```diff
@@ -281,6 +282,12 @@ watch(() => route.params, async (params) => {
                             </span>
                         </div>
                     </template>
+                    <template #menu-item--format="{item}">
+                        <div class="menu-item">
+                            <!--TODO: Currency should be changed to real data.-->
+                            <span>{{ item.label }}</span> <span class="selected-item-postfix">(Currency: ₩KRW)</span>
+                        </div>
+                    </template>
```
Comment (sulmoJ) [inline apps/web/src/services/cost-explorer/CostExplorerLNB.vue:289]: It was eventually added ㅋㅋ

Comment (piggggggggy) [reply apps/web/src/services/cost-explorer/CostExplorerLNB.vue:289]: hahaha

**Pull request merged**